### DISCUSSION TITLE
Admin-Rechte für custom_homematic Kommunikation benötigt / Warum?

### DISCUSSION THREAD

**Discussion by Timmmy001** [SukramJ/aiohomematic#1406] [General]
Hallo Zusammen,
bei der Einrichtung von custom_homematic für die Anbindung von HA an einen homematic-raspi, war ich überrascht dass das Konto bei homematic AdminRechte benötigt.
Auch ein nachträgliches entziehen der Rechte funktioniert leider nicht.

Ich komme aus der FHEM-Welt und habe da die gleiche Anbindung in "Grün"... und dort braucht es **nur Benutzerrechte**.
Aus Sicherheitsaspekten würde ich daher lieber wenige Rechte geben und frage mich, was bei dieser hier Implementierung so anderes ist.

Hat hier jemand Antworten?

**Comment by SukramJ** [comment]
> Ich komme aus der FHEM-Welt und habe da die gleiche Anbindung in "Grün"... und dort braucht es nur Benutzerrechte.

Das wird wohl nicht so sein.

Da Methoden (ReGa.runScript) aufgerufen werden, die Admin-Rechte benötigen, brauchst Du halt einen Benutzer mit Admin-Rechten.

**Comment by Timmmy001** [comment]
Ähm, tuschldigung... aber es ist so und ich kann die Geräte über FHEM steuern.

![grafik](https://github.com/danielperna84/hahomematic/assets/39485318/6cace65c-6bb8-472d-80c9-e148c422652c)
![grafik](https://github.com/danielperna84/hahomematic/assets/39485318/d0c9b241-f25b-4074-b254-34cb912ee445)

**Reply by SukramJ** [reply]
Wenn Du mir jetzt erklären willst wie eine CCU Anbindung funktioniert, dann schau die doch bitte zuerst den Source-Code an!
Natürlich wird der Callback verwendet aber wir verwenden unter anderem auch `ReGa.runScript`.

**Reply by Timmmy001** [reply]
Entspannung...
Ich habe lediglich verglichen, dass bei scheinbar gleicher Funktionalität, der eine trotzdem nur Benutzerrechte benötigt und
gegen das "Das wird wohl nicht so sein." einen Beweis geliefert...

**Reply by Timmmy001** [reply]
Nichts desto Trotz würde ich eine Implementierung mit weniger Rechtsanspruch bevorzugen.

**Reply by danielperna84** [reply]
FHEM holt sich nicht so viele Daten von der CCU wie diese Integration. Hier werden Gerätenamen, Räume usw. geholt, welche nicht über die XML-RPC API zu Verfügung gestellt werden.

Die Kommunikation mit den Sensoren und Aktoren an sich würde auch hier ohne Admin-Rechte gehen. Aber wegen des Comforts der schönen Benamung und Zuordnung der Geräte (was eine Vielzahl der User wünscht), braucht es eben die höheren Berechtigungen.

**Reply by Timmmy001** [reply]
Ok, das ist eine valide Erklärung.
Vielen Dank (+Erstellung des Moduls)!
